File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLInputPrivacyValueDisallowContacts.java
package com.github.badoualy.telegram.tl.api;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLInputPrivacyValueDisallowContacts extends TLAbsInputPrivacyRule {

    public static final int CONSTRUCTOR_ID = 0xba52007;

    private final String _constructor = "inputPrivacyValueDisallowContacts#ba52007";

    public TLInputPrivacyValueDisallowContacts() {
    }

    @Override
    public String toString() {
        return _constructor;
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLInputPrivacyValueDisallowContacts.kt
package com.github.badoualy.telegram.tl.api

/**
 * inputPrivacyValueDisallowContacts#ba52007
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLInputPrivacyValueDisallowContacts : TLAbsInputPrivacyRule() {
    private val _constructor: String = "inputPrivacyValueDisallowContacts#ba52007"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLInputPrivacyValueDisallowContacts) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0xba52007.toInt()
    }
}
